src/libxrpl/protocol/STTx.cpp

```diff
@@ -227,6 +229,32 @@ STTx::checkSign(
     return Unexpected("Internal signature check failure.");
 }
 
+Expected<void, std::string>
+STTx::checkBatchSign(
+    RequireFullyCanonicalSig requireCanonicalSig,
+    Rules const& rules) const
+{
+    try
+    {
+        STArray const& signers{getFieldArray(sfBatchSigners)};
+        for (auto const& signer : signers)
+        {
+            Blob const& signingPubKey = signer.getFieldVL(sfSigningPubKey);
+            auto const result = signingPubKey.empty()
+                ? checkBatchMultiSign(signer, requireCanonicalSig, rules)
+                : checkBatchSingleSign(signer, requireCanonicalSig);
+
+            if (!result)
+                return result;
+        }
+        return {};
+    }
+    catch (std::exception const&)
+    {
```

Comment:
It might be but the original implementation of `STTx::checkSign` does not catch and Rethrow. So if you have good suggestion to why I should deviate from the original implementation then I would love to hear it.

Reply:
I'm new to this piece of code. I feel the exception message is useful for debugging purposes. I don't feel strongly about it. 

tbh I'm not entirely sure if this rethrown exception is bubbled out into the response at all.